### DISCUSSION TITLE
[WASM] Simple WASM app with Multithreading activated starts with error: "Failed to log "ReadPackageFileAsync ... System.InvalidOperationException: Please use dedicated worker for working with JavaScript interop."

### DISCUSSION THREAD

**Discussion by tobizk04** [unoplatform/Uno.Wasm.Bootstrap#869] [Q&A]
If I create a new uno platform app with the wizard and start the wasm app I get the following interop errors.

![image](https://github.com/unoplatform/Uno.Wasm.Bootstrap/assets/165020047/63670ba1-0800-4af4-8fe5-0a34118b09a8)

To reproduce:
Create a new uno platform app with following featurs and multithreading active:
```

<UnoFeatures>
  Material;
  Dsp;
  Hosting;
  Toolkit;
  Logging;
  Mvvm;
  Configuration;
  Http;
  Serialization;
  Localization;
  Navigation;
  ThemeService;
</UnoFeatures>

<WasmShellEnableThreads>true</WasmShellEnableThreads>
<WasmShellPThreadsPoolSize>8</WasmShellPThreadsPoolSize>

```
```

<TargetFrameworks>
  net8.0-windows10.0.19041;
  net8.0-browserwasm;
</TargetFrameworks>

```

This is the solution:
-> [UnoApp8.zip](https://github.com/user-attachments/files/16123855/UnoApp8.zip)

**Is this an UNO bug?**

The error trace starts with.
at Uno.Extensions.Logging.WebAssembly.WebAssemblyConsoleLogger.WriteMessage(LogLevel logLevel, String logName, Int32 eventId, String message, Exception exception) in /_/src/Uno.Extensions.Logging.WebAssembly.Console/WebAssemblyConsoleLogger.cs:line 99

And there is no custom interop code implemented.

Should I create an bug issue for this?
I think it could be handled with main thread is used until .net runtime supports multithreading javascript interop.

What do you mean?

**Comment by jeromelaban** [comment]
Thanks for the question. Let's keep the discussion flowing there: https://github.com/unoplatform/uno/discussions/17365.
